scipy/sparse/compressed.py

```diff
@@ -784,7 +784,8 @@ def _get_submatrix(self, major=None, minor=None, copy=False):
         j0, j1 = _process_slice(minor, N)
 
         if i0 == 'empty' or j0 == 'empty':
-            out = np.zeros([0, 0], dtype=self.dtype)
+            out = np.zeros(shape=(0, self.shape[1]),
+                           dtype=self.dtype)
```

Comment:
I don't think this covers the general case that includes a potentially empty slice along the second axis (or along both axes at once). I think if the empty slice was along the second axis it should be `shape=(self.shape[0], 0)` or if along both axes then `shape=(0, 0)` as it was before. 

I may be able to take a look tomorrow if you don't get a chance to try it before then.